src/daemon/collectd.c

```diff
@@ -91,13 +81,20 @@ static int init_hostname(void) {
   struct addrinfo *ai_list;
   int status;
 
+  long hostname_len = sysconf(_SC_HOST_NAME_MAX);
+  if (hostname_len == -1) {
+    hostname_len = NI_MAXHOST;
+  }
+  char hostname[hostname_len];
+  hostname_set(hostname);
```

Comment:
You're calling `hostname_set()` with an uninitialized buffer here. I think this needs to be removed – the code below is taking care of calling `hostname_set()` once the buffer is initialized.